File: checkered_board.rb
#!/usr/bin/env ruby
# http://www.codewars.com/kata/checkered-board

def checkered_board(size)
  white_square, black_square = "■", "□"
  (0...size).each do |c|
    (0...size).each do |r|
      if r % 2 == 0
        print c % 2 == 0 ? white_square : black_square
        print " "
      else
        print c % 2 == 0 ? black_square : white_square
        print " "
      end
    end
    puts
  end
end

checkered_board(5)

File: checkered_board.swift
#!/usr/bin/env swift
// http://www.codewars.com/kata/checkered-board

func checkeredBoard(size: Int) {
  let whiteSquare = "■"
  let blackSquare = "□"
  var rowEven = ""
  var rowOdd = ""
  for i in 0..<size {
    rowEven += i % 2 == 0 ? whiteSquare : blackSquare
    rowEven += " "
  }
  for i in 0..<size {
    rowOdd += i % 2 == 0 ? blackSquare : whiteSquare
    rowOdd += " "
  }
  for i in 0..<size {
    i % 2 == 0 ? print(rowEven) : print(rowOdd)
  }
}

checkeredBoard(8)
